Source/ee/VuBasicBlock.cpp

```diff
@@ -199,6 +261,46 @@ void CVuBasicBlock::CompileRange(CMipsJitter* jitter)
 			jitter->PullRel(offsetof(CMIPS, m_State.nCOP2VI[integerBranchDelayInfo.regIndex]));
 		}
 
+		if(address == m_begin)
+		{
+			if(IsConditionalBranch(opcodeLo))
+			{
+				/*
+				    jit code needs to do something like follows:
+
+				    if(branchLoOps.readI0 == m_State.savedNextBlockIntRegIdx || branchLoOps.readI1 == m_State.savedNextBlockIntRegIdx)
+				    {
+				        m_State.nCOP2VI[m_State.savedNextBlockIntRegIdx] = m_State.savedNextBlockIntRegValTemp;
+				    }
+				 */
+				auto branchLoOps = arch->GetAffectedOperands(&m_context, address, opcodeLo);
+
+				auto resetDelayedValLabel = jitter->CreateLabel();
+
+				jitter->PushRel(offsetof(CMIPS, m_State.savedNextBlockIntRegIdx));
+				jitter->PushCst(branchLoOps.readI0);
+				jitter->BeginIf(Jitter::CONDITION_EQ);
```

Comment:
Would it work to do something like that?
```
jitter->PushRel(offsetof(CMIPS, m_State.savedNextBlockIntRegIdx));
jitter->PushCst(branchLoOps.readI0);
jitter->Cmp(CONDITION_EQ);
jitter->PushRel(offsetof(CMIPS, m_State.savedNextBlockIntRegIdx));
jitter->PushCst(branchLoOps.readI1);
jitter->Cmp(CONDITION_EQ);
jitter->Or();
jitter->PushCst(0);
jitter->BeginIf(Jitter::CONDITION_NE);
{
    //... Logic here
}
jitter->EndIf();

Reply:
ah yes - I missed the Cmp instruction. That's much more expressive.

Reply:
changed and squashed.